### PR TITLE
fix: added zod for type safe parsing

### PR DESCRIPTION
**Pull request by markusahlstrand**
This is freakin cool. Zod finally solves the issue with unsafe types runtime without adding more cruft to the code.

As the durable object storage just handles strings I would really like to have a safe way of getting and storing data there. This is a first POC and the next step would be make a generic wrapper for the storage.

### CODE REVIEW

**Comment by markusahlstrand** [comment]
Ahh.. and we are already using Zod as it's part of tRPC, so it doesn't add any new packages.

**Comment by danielgent** [comment]
> Ahh.. and we are already using Zod as it's part of tRPC, so it doesn't add any new packages.

But you should add `zod` as a dependency @markusahlstrand ! :smiling_face_with_tear: 

PNPM doesn't allow importing from dependencies of dependencies :smile: 

There's an error in the test CI/CD with types, maybe related to this?

**Comment by markusahlstrand** [comment]
The error is that Tsoa doesn't understand the type.. Really annoying

**Comment by markusahlstrand** [comment]
> 

Added Zod to the dependencies now :)

**Comment by markusahlstrand** [comment]
So.. I'll write what I wrote in slack here as well:
We are duplicating the schema, but as Typescript is converting the types in a typesafe fashion it will complain compile time if it doesn't match. So.. a bit verbose but typesafe. Hopefully we can patch TSOA to make it work even better.

File: package.json [hunk 1]
```diff
@@ -55,7 +55,8 @@
     "mime": "^3.0.0",
     "trpc-durable-objects": "^1.3.3",
     "tsoa": "^5.1.1",
-    "tsoa-workers": "^1.3.6"
+    "tsoa-workers": "^1.3.6",
+    "zod": "^3.21.4"
```
Comment (danielgent) [inline package.json:59]: I suppose the issue here could be that `tsoa` starts using a different version of `zod` and cloudflare workers are sensitive to bundle size? :thinking: 

Probably not an issue though

Comment (markusahlstrand) [reply package.json:59]: Yepp, I had this in the knex-adaptor I wrote.. I wonder if we should add it as a peer-dependency?

Comment (markusahlstrand) [reply package.json:59]: Peer-dependency doesn't seem right.. Need to keep track of this..

File: test/models/User.spec.ts [hunk 2]
```diff
@@ -181,6 +181,8 @@ describe("User", () => {
               return JSON.stringify({
                 id: "id",
                 name: "Test",
+                email: "test@example.com",
+                tenantId: "tenantId",
```
Comment (danielgent) [inline test/models/User.spec.ts:185]: Cool. Zod pointed this out?

File: package.json (hunk 1 above)
Comment (danielgent) [reply package.json:59]: :shrug: Pin the versions?

I suppose the ideal would be CI/CD tooling to monitor the final bundle and tell us about duplicates

**Review by danielgent — APPROVED** [review]

File: src/models/User.ts [hunk 3]
```diff
@@ -52,15 +52,14 @@ const PROFILE_FIELDS = [
...
-async function getProfile(
-  storage: DurableObjectStorage,
-): Promise<Profile | null> {
-  const profilesString = await storage.get<string>(StorageKeys.profile);
-  if (!profilesString) {
+async function getProfile(storage: DurableObjectStorage) {
+  const jsonData = await storage.get<string>(StorageKeys.profile);
+
+  try {
+    return ProfileSchema.parse(JSON.parse(jsonData as string));
```
Comment (danielgent) [inline src/models/User.ts:59]: that is cool!

Do we want to just swallow the error though? What would an error mean here?

**Pull request merged**